src/main/java/org/gridsuite/shortcircuit/server/repositories/ShortCircuitAnalysisResultRepository.java

```diff
@@ -248,52 +254,156 @@ public Optional<ShortCircuitAnalysisResultEntity> findResultsWithLimitViolations
-        if (faultResultsPage.hasContent()) {
-            appendLimitViolationsAndFeederResults(faultResultsPage);
+
+        Page<FaultResultRepository.EntityId> uuidPage = faultResultRepository.findBy(specification, q ->
+                q.project(FaultResultEntity.Fields.faultResultUuid)
+                        .as(FaultResultRepository.EntityId.class)
+                        .sortBy(modifiedPageable.getSort())
+                        .page(modifiedPageable)
+        );
+
+        if (uuidPage.hasContent()) {
+            List<UUID> faultResultsUuids = uuidPage
+                    .map(FaultResultRepository.EntityId::getFaultResultUuid)
+                    .toList();
+            // Then we fetch the main entities data for each UUID
+            List<FaultResultEntity> faultResults = faultResultRepository.findAllByFaultResultUuidIn(faultResultsUuids);
+            faultResults.sort(Comparator.comparing(fault -> faultResultsUuids.indexOf(fault.getFaultResultUuid())));
+            Page<FaultResultEntity> faultResultsPage = new PageImpl<>(faultResults, modifiedPageable, uuidPage.getTotalElements());
+
+            // then we append the missing data, and filter some of the Lazy Loaded collections
+            appendLimitViolationsAndFeederResults(faultResultsPage, childrenSort, resourceFilters);
+
+            return faultResultsPage;
+        } else {
+            return Page.empty();
         }
-        return faultResultsPage;
     }
 
-    private void appendLimitViolationsAndFeederResults(Page<FaultResultEntity> pagedFaultResults) {
+    private void appendLimitViolationsAndFeederResults(Page<FaultResultEntity> pagedFaultResults,
+                                                       Optional<Sort.Order> childrenSort,
+                                                       List<ResourceFilter> resourceFilters) {
         // using the Hibernate First-Level Cache or Persistence Context
         // cf.https://vladmihalcea.com/spring-data-jpa-multiplebagfetchexception/
         if (!pagedFaultResults.isEmpty()) {
             List<UUID> faultResultsUuids = pagedFaultResults.stream()
                     .map(FaultResultEntity::getFaultResultUuid)
-                    .collect(Collectors.toList());
+                    .toList();
+
+            Specification<FaultResultEntity> specification = faultSpecBuilder.buildFeedersSpecification(faultResultsUuids, resourceFilters);
+            faultResultRepository.findAll(specification);
+
             faultResultRepository.findAllWithLimitViolationsByFaultResultUuidIn(faultResultsUuids);
-            faultResultRepository.findAllWithFeederResultsByFaultResultUuidIn(faultResultsUuids);
+
+            sortFeeders(pagedFaultResults, childrenSort);
+        }
+    }
+
+    private void sortFeeders(Page<FaultResultEntity> pagedFaultResults, Optional<Sort.Order> childrenSort) {
```

Comment:
Btw, the `Page` already contains the children sort

Reply:
Yes mb, but don't we want to keep the children sort ? Like, we want to filter from higher to lower value, don't we want to have the highest value on the first page ?